website/static/js/share/utils.js

```diff
@@ -190,30 +228,269 @@ utils.removeFilter = function(vm, filter){
     utils.search(vm);
 };
 
-utils.arrayEqual = function(a, b) {
+/* Tests array equality */
+utils.arrayEqual = function (a, b) {
     return $(a).not(b).length === 0 && $(b).not(a).length === 0;
 };
 
-utils.loadStats = function(vm){
-    vm.statsLoaded(false);
+utils.addFiltersToQuery = function (query, filters) {
+    if (filters) {
+        filters.forEach(function (filter) {
+            query = utils.filteredQuery(query, filter.filter);
+        });
+    }
+    return query;
+};
 
+/* Loads the raw and normalized data for a specific result */
+utils.loadRawNormalized = function(result){
+    var nonJsonErrors = function(xhr) {
```

Comment:
Thanks for the clarification. Are only error pages returned as HTML? If so, then shouldn't you check `>= 400` rather than `> 200`?

A code comment here would be helpful, since the API's behavior isn't obvious.